### ISSUE TITLE
[a11y][iOS] bottomNavigationBar cannot be navigated using Switch Control

### ISSUE THREAD

**Issue by ChristianEdwardPadilla** [flutter/flutter#160033]
### Steps to reproduce

1. Enable Switch Control in iOS settings app (https://support.apple.com/en-us/119835).
2. Using the repro app below, attempt to navigate between `NavigationDestination`s within the `NavigationBar`.

### Expected results

It is possible to navigate between `NavigationDestination` tabs.

### Actual results

Focus seems to be stuck on a single `NavigationDestination` tab.

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: ThemeData(useMaterial3: false),
      title: 'Flutter Demo',
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;
  int _selectedIndex = 0;

  void _incrementCounter() {
    setState(() {
      _counter++;
    });
  }

  void _changeIndex(int index) {
    setState(() {
      _selectedIndex = index;
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title, style: TextStyle(fontFamily: 'ProductSans')),
      ),
      body: Center(
        child: Column(
          children: [
            Text(
              'Button tapped $_counter time${_counter == 1 ? '' : 's'}.',
              // This optional key is used to uniquely identify this widget in
              // the integration test at: test_driver/tap_test.dart.
              key: Key('CountText'),
              style: TextStyle(fontSize: 24),
            ),
            MaterialButton(
              onPressed: () {
                print('Button tapped');
              },
              color: Colors.blue,
              child: Text('Test button 1'),
            ),
            MaterialButton(
              onPressed: () {
                print('Button tapped');
              },
              color: Colors.blue,
              child: Text('Test button 2'),
            ),
            MaterialButton(
              onPressed: () {
                print('Button tapped');
              },
              color: Colors.blue,
              child: Text('Test button 3'),
            ),
          ],
        ),
      ),
      bottomNavigationBar: NavigationBar(
        onDestinationSelected: (index) {
          print('Selected destination $index');
          _incrementCounter();
          _changeIndex(index);
        },
        selectedIndex: _selectedIndex,
        destinations: [
          NavigationDestination(label: 'one', icon: Icon(Icons.home)),
          NavigationDestination(label: 'two', icon: Icon(Icons.home)),
          NavigationDestination(label: 'three', icon: Icon(Icons.home)),
        ],
      ),
    );
  }
}

```

</details>

### Screenshots or Video

<details open>
<summary>Video demonstration</summary>

https://github.com/user-attachments/assets/7e246174-37fd-4d5b-a8be-d21c4e9a449b

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel google3)
    • Framework revision 1d60fc7843 (2 days ago), 2024-12-08T00:00:00.000
    • Engine revision 82d7c1b2c2
    • Dart version c7e47c6c5d

[✓] Xcode - develop for iOS and macOS (Xcode 16.0)
    • Xcode at /Applications/Xcode_16.0.0.app/Contents/Developer
    • Build 16A242d
```

</details>

Google bug tracker: b/382363824.

**Comment by maheshj01** [comment]
Seems related to  https://github.com/flutter/flutter/issues/139235, https://github.com/flutter/flutter/issues/44077

**Comment by danagbemava-nc** [comment]
Reproducible using the steps outlined above. 

<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, 3.24.5, on macOS 15.1.1 24B91 darwin-arm64, locale en-US)
    • Flutter version 3.24.5 on channel stable at /Users/deanli/dev/stable
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision dec2ee5c1f (4 weeks ago), 2024-11-13 11:13:06 -0800
    • Engine revision a18df97ca5
    • Dart version 3.5.4
    • DevTools version 2.37.3
```

```
[!] Flutter (Channel master, 3.27.0-1.0.pre.778, on macOS 15.1.1 24B91 darwin-arm64, locale en-US)
    • Flutter version 3.27.0-1.0.pre.778 on channel master at /Users/deanli/dev/master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 9c86062f94 (5 hours ago), 2024-12-10 22:03:41 -0500
    • Engine revision a8f7556472
    • Dart version 3.7.0 (build 3.7.0-232.0.dev)
    • DevTools version 2.41.0
```
</details>
